mitmproxy/addons/allowremote.py

```diff
@@ -0,0 +1,16 @@
+import ipaddress
+from mitmproxy import ctx
+
+
+class AllowRemote:
+    def clientconnect(self, layer):
+        address = layer.client_conn.address
+
+        if not (ctx.options.allow_remote or
+                ipaddress.ip_address(address[0]).is_private or
```

Comment:
`is_private` could be anything that is considered *private* by the IANA. I think this is a bit too much, and we should restrict it to only allow the local network (netmask in IPv4 or link-local address in IPv6).
Thoughts?